File: 100_Same_Tree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        # If both trees are empty, they are the same
        if not p and not q:
            return True
        # If one tree is empty and the other is not, they are not the same
        if not p or not q:
            return False
        # Check if the values of the current nodes are the same
        if p.val != q.val:
            return False
        # Recursively check left and right subtrees
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)

File: 100_Same_Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val === undefined ? 0 : val);
 *     this.left = (left === undefined ? null : left);
 *     this.right = (right === undefined ? null : right);
 * }
 */

/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function (p, q) {
  // If both trees are empty, they are the same
  if (!p && !q) return true;

  // If one tree is empty and the other is not, they are not the same
  if (!p || !q) return false;

  // Check if the current nodes have the same value
  if (p.val !== q.val) return false;

  // Recursively check left and right subtrees
  return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
};
